c++/src/kj/compat/http.c++

```diff
@@ -3373,8 +3392,9 @@ private:
           KJ_IF_SOME(m, maxSize) {
             // This is only non-null for `receive` calls, so we must be decompressing. We don't want
             // the decompressed message to OOM us, so let's make sure it's not too big.
-            KJ_REQUIRE(totalBytesProcessed < m,
-                "Decompressed WebSocket message is too large");
+            if (totalBytesProcessed > m) {
+              kj::throwRecoverableException(KJ_EXCEPTION(FAILED, kj::str("Decompressed WebSocket message is too large: ", totalBytesProcessed, " > ", m)));
+            }
```

Comment:
Yeah makes sense, it felt a bit hacky but I wasn't sure what's the appropriate way to go about it.
It might be reasonable to return OneOf<..., HttpHeaders::ProtocolError> just to be a little more forward compatible, what do you think?

Reply:
Yeah the manual formatting was because I wanted to return the description but I agree it feels hacky,.
I switched ProcesMessage to return OneOf<..., ProtocolError> which seemed appropriate.
What do you think about how it looks now?